crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -129,6 +158,18 @@ where
         info!(target: COMPONENT, num_txs, "Building a transaction batch");
         debug!(target: COMPONENT, txs = %format_array(txs.iter().map(|tx| tx.id().to_hex())));
 
+        let dangling_notes = self.find_dangling_notes(&txs).await;
+        if !dangling_notes.is_empty() {
+            let missing_notes = match self.store.get_missing_notes(&dangling_notes).await {
+                Ok(notes) => notes,
+                Err(err) => return Err(BuildBatchError::GetMissingNotesRequestError(err, txs)),
+            };
+
+            if !missing_notes.is_empty() {
+                return Err(BuildBatchError::UnauthenticatedNotesNotFound(missing_notes, txs));
+            }
```

Comment:
As discussed offline, let's invert this and use `GetNotesById` to retrieve notes which are already in the store.

Let's also create an issue to make a specialized endpoint later which would return both note IDs and note inclusion paths.

Reply:
I've created an issue for this: https://github.com/0xPolygonMiden/miden-node/issues/395